Branch-SDK/Branch-SDK/BNCSystemObserver.m

```diff
@@ -121,13 +121,16 @@ + (NSString *)getDeviceName {
 }
 
 + (NSNumber *)getUpdateState {
-    NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];    NSFileManager *manager = [NSFileManager defaultManager];
-    NSDictionary* attrs = [manager attributesOfItemAtPath:bundleRoot error:nil];
-    if ((int)([[attrs fileCreationDate] timeIntervalSince1970]/(60*60*24)) == (int)([[attrs fileModificationDate] timeIntervalSince1970]/(60*60*24))) {
-        return nil;
-    } else {
+
+    NSString *storedAppVersion = [BNCPreferenceHelper getAppVersion];
+    NSString *currentAppVersion = [BNCSystemObserver getAppVersion];
+
+    if (storedAppVersion == nil ||
+        ([storedAppVersion compare:currentAppVersion options:NSNumericSearch] == NSOrderedAscending)) {
+        [BNCPreferenceHelper setAppVersion:currentAppVersion];
```

Comment:
but what about the case when they first integrate Branch on a prexisting app with a large install base? We can't show all updates as installs

The problem with today is that the app developer had 2k actual itunesconnect downloads and 200k upgrades. We counted 20k incorrectly as installs instead of updates